test/e2e/reconciler/cache/replication_test.go

```diff
@@ -60,109 +61,264 @@ type testScenario struct {
 // scenarios all test scenarios that will be run against in-process and standalone cache server
 var scenarios = []testScenario{
 	{"TestReplicateAPIExport", replicateAPIExportScenario},
+	{"TestReplicateAPIResourceSchema", replicateAPIResourceSchemaScenario},
+	{"TestReplicateAPIExportNegative", replicateAPIExportNegativeScenario},
+}
+
+// baseScenario an auxiliary struct that is used by replicateResourceScenario
+type baseScenario struct {
+	server framework.RunningServer
+
+	resourceName string
+	resourceKind string
+
+	createSourceResource        func(logicalcluster.Name) error
+	getSourceResource           func(logicalcluster.Name) (interface{}, error)
```

Comment:
I think we talked about this before, but there's nothing more specific that this can be, like `runtime.Object`? and we don't want to use generics, because?

Reply:
in this particular case, we don't need strong types, we operate on ` meta.Accessor`

Reply:
Can we use `meta.Accessor` then?

Reply:
sure, will change it.

Reply:
changed to `runtime.Object` because I've realized we need full object for comparison.